Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,230 @@
+
+### Opting into certain regex features
+
+We intend for the compiler parsing logic to completely parse [the PCRE syntax][PCRE], however types conforming to `RegexLiteralProtocol` may not be able to handle the full regex grammar. To signal the features that they support, conforming types will be able to annotate builder functions with availability; which allows them to communicate the features they don't intend on supporting, in addition to the features they may support in the future.
+
+This also would give the standard library the flexibility to roll out new regex features over time.
+
+### Impact of using `/` as the delimiter
+
+#### On custom prefix and postfix operators using the `/` character
+
+Choosing `/` as the delimiter means there will be parsing ambiguity with any user-defined prefix and postfix operators containing the `/` character. For example, code such as the following poses an issue:
+
+```swift
+let x = /0; let y = 1/
+```
+
+Should this be considered to be two `let` bindings, with each initialization expression using prefix and postfix `/` operators, or is it a single regex literal?
+
+This also extends to more generally to prefix and postfix operators containing the `/` character, e.g:
+
+```swift
+let x = </<0; let y = 1</<
+```
+Is this a regex literal `/<0; let y = 1</` with a prefix and postfix `<` operator applied, or two `let` bindings each using prefix and postfix `</<` operators?
+
+There are no easy ways of resolving these ambiguities, therefore a regex literal parsed with `/` delimiters will likely need to be introduced under a new language version mode, along with a deprecation of prefix and postfix operators that use the `/` character. In the new language version mode, the above examples will be parsed as regex literals.
+
+#### On custom infix operators using the `/` character
+
+There is also a conflict for infix operators containing `/` in cases where whitespace isn't used, for example:
+
+```swift
+x+/y/+z
+```
+
+Should the operators be parsed as `+/` and `/+` respectively, or should this be parsed as `x + /y/ + z`?
+
+In this case, things can be more readily disambiguated by the user inserting additional whitespace. We therefore could potentially continue to parse `x+/y/+z` as a binary operator chain, and require additional whitespace to interpret `/y/` as a regex literal. However, depending on the source compatibility impact, for consistency with the above proposed deprecation of prefix and postfix operators containing `/`, it may be prudent to also deprecate infix operators containing `/`. This would let us have a consistent rule, and would simplify the parsing logic required. Note though this deprecation would not affect the existing division operator `/`.
```

Comment:
Ah whoops it's totally not possible to deprecate infix operators containing `/` as we have the `/=` operator...